common/data_refinery_common/message_queue.py

```diff
@@ -77,26 +77,28 @@ def send_job(job_type: Enum, job, is_dispatch=False) -> bool:
                 job_type.value,
                 job.id)
 
-    # Smasher doesn't need to be on a specific instance since it will
-    # download all the data to its instance anyway.
-    if isinstance(job, ProcessorJob) and job_type not in [ProcessorPipeline.SMASHER, ProcessorPipeline.QN_REFERENCE]:
-        # Make sure this job goes to the correct EBS resource.
-        # If this is being dispatched for the first time, make sure that
-        # we store the currently attached index.
-        # If this is being dispatched by the Foreman, it should already
-        # have an attached volume index, so use that.
-        if job.volume_index is None:
-            job.volume_index = get_volume_index()
-            job.save()
-        nomad_job = nomad_job + "_" + job.volume_index + "_" + str(job.ram_amount)
-    elif isinstance(job, SurveyJob):
-        nomad_job = nomad_job + "_" + str(job.ram_amount)
-    elif isinstance(job, DownloaderJob):
-        nomad_job = nomad_job + "_" + str(job.ram_amount)
-
     # We only want to dispatch processor jobs directly.
     # Everything else will be handled by the Foreman, which will increment the retry counter.
     if is_processor or is_dispatch or (not settings.RUNNING_IN_CLOUD):
+
+        # Smasher doesn't need to be on a specific instance since it will
+        # download all the data to its instance anyway.
+        if isinstance(job, ProcessorJob) and job_type not in [ProcessorPipeline.SMASHER, ProcessorPipeline.QN_REFERENCE]:
+            # Make sure this job goes to the correct EBS resource.
+            # If this is being dispatched for the first time, make sure that
+            # we store the currently attached index.
+            # If this is being dispatched by the Foreman, it should already
+            # have an attached volume index, so use that.
+            if job.volume_index is None:
+                job.volume_index = get_volume_index()
+                job.save()
+            nomad_job = nomad_job + "_" + job.volume_index + "_" + str(job.ram_amount)
+        elif isinstance(job, SurveyJob):
+            nomad_job = nomad_job + "_" + str(job.ram_amount)
+        elif isinstance(job, DownloaderJob):
+            volume_index = job.volume_index if settings.RUNNING_IN_CLOUD else "0"
+            nomad_job = nomad_job + "_" + volume_index + "_" + str(job.ram_amount)
```

Comment:
I think moving this code block would introduce a bug because we'd start queuing downloader jobs immediately. In fact, I don't think this is where we should be setting the volume_index. I think we should have the foreman decide which volume to put downloader jobs on as it dispatches them.

Reply:
Ok, as we discussed in person I now see why this is in fact correct: this will only fire when `is_dispatch=True` which will be the case when the Foreman is actually queuing downloader jobs.